### ISSUE TITLE
Unnumbered headers are numbered in TOC in HTML output when `--number-sections` and `--number-offset` are used

### ISSUE THREAD

**Issue by fbristow** [jgm/pandoc#6346]
Summary
========

I'm using `{-}` on some of my headings to make them unnumbered, but when I use `--number-sections` and `--number-offset=`, the TOC entries for unnumbered sections have numbers beside them in HTML output.

pandoc version: 2.9.2.1

Sample input
==========

```markdown
First {-}
=====

Second {.unnumbered}
======

Third
=====
```

Command used
============

```bash
pandoc input.md --number-sections --toc --standalone --number-offset=3 -t html
```

Actual output
==========
**Note**: Header is omitted for brevity.

```html
<body>
<nav id="TOC" role="doc-toc">
<ul>
<li><a href="#first"><span class="toc-section-number">3</span> First</a></li>
<li><a href="#second"><span class="toc-section-number">3</span> Second</a></li>
<li><a href="#third"><span class="toc-section-number">4</span> Third</a></li>
</ul>
</nav>
<section id="first" class="unnumbered" data-number="">
<h1 class="unnumbered" data-number="3">First</h1>
</section>
<section id="second" class="unnumbered" data-number="">
<h1 class="unnumbered" data-number="3">Second</h1>
</section>
<section id="third" data-number="1">
<h1 data-number="4"><span class="header-section-number">4</span> Third</h1>
</section>
</body>
```

Expected output
=============

I expect the same/similar output as above, but without numbers in the TOC for the entries marked as `.unnumbered`. Note that the `<section>` tags *do not* appear when `--number-offset=` is unset.

**Comment by jgm** [comment]
This was fixed already by 83880b0dbc318703babfbb6905b1046fa48f1216

**Comment by fbristow** [comment]
Thanks again, @jgm. Thank you also for making it so easy to build a `deb` package, it was trivial to install from `master` to get this fix.
